src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/TypeGVMEntriesNode.cs

```diff
@@ -0,0 +1,120 @@
+        public override IEnumerable<CombinedDependencyListEntry> SearchDynamicDependencies(List<DependencyNodeCore<NodeFactory>> markedNodes, int firstNode, NodeFactory context)
+        {
+            return Array.Empty<CombinedDependencyListEntry>();
+        }
+
+        public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFactory context)
+        {
+            if (_staticDependencies == null)
+            {
+                _staticDependencies = new DependencyList();
+
+                foreach(var entry in ScanForGenericVirtualMethodEntries())
+                    _staticDependencies.AddRange(GenericVirtualMethodTableNode.GetGenericVirtualMethodImplementationDependencies(context, entry.CallingMethod, entry.ImplementationMethod));
+
+                foreach (var entry in ScanForInterfaceGenericVirtualMethodEntries())
+                    _staticDependencies.AddRange(InterfaceGenericVirtualMethodTableNode.GetGenericVirtualMethodImplementationDependencies(context, entry.CallingMethod, entry.ImplementationType, entry.ImplementationMethod));
+            }
+
+            return _staticDependencies;
+        }
+
+        public static bool TypeNeedsGVMTableEntries(TypeDesc type)
+        {
+            // Only non-interface deftypes can have entries for their GVMs in the GVM hashtables.
+            // Interface GVM entries are computed for types that implemenent the interface (not for the interface on its own)
+            if(!type.IsDefType || type.IsInterface)
+                return false;
+
+            return type.HasGenericVirtualMethod();
+        }
+
+        public IEnumerable<TypeGVMEntryInfo> ScanForGenericVirtualMethodEntries()
+        {
+            foreach (var method in _associatedType.GetMethods())
+            {
+                if (!method.IsVirtual || !method.HasInstantiation)
+                    continue;
+
+                MethodDesc slotDecl = MetadataVirtualMethodAlgorithm.FindSlotDefiningMethodForVirtualMethod(method);
+                Debug.Assert(slotDecl != null);
+                yield return new TypeGVMEntryInfo(slotDecl, method, null);
+            }
+        }
+
+        public IEnumerable<TypeGVMEntryInfo> ScanForInterfaceGenericVirtualMethodEntries()
+        {
+            foreach (var iface in _associatedType.RuntimeInterfaces)
+            {
+                foreach (var method in iface.GetMethods())
+                {
+                    if (!method.HasInstantiation || method.Signature.IsStatic)
+                        continue;
+
+                    MethodDesc slotDecl;
+                    TypeDesc currentType = _associatedType;
+                    do
+                    {
+                        slotDecl = currentType.ResolveInterfaceMethodToVirtualMethodOnType(method);
+                        currentType = currentType.BaseType;
+                    }
+                    while (slotDecl == null && currentType != null);
+
+                    Debug.Assert(slotDecl != null);
```

Comment:
I will remove this assert. I think it is wrong (ex: abstract type that doesn't implement all interface methods of all implemented interfaces). I'll also add a test for it in my GVM unit test (which i'll send later in another PR)